frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/tabs/vm-settings-tab/provision-source.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { FormSelect, FormSelectOption, Button, ButtonVariant } from '@patternfly/react-core';
+import { FormSelectPlaceholderOption } from '../../../form/form-select-placeholder-option';
+import { ProvisionSource } from '../../../../constants/vm/provision-source';
+import { FormFieldRow } from '../../form/form-field-row';
+import { FormField, FormFieldType } from '../../form/form-field';
+import { iGetFieldValue } from '../../selectors/immutable/field';
+import { VMSettingsField } from '../../types';
+import { getPlaceholder } from '../../utils/renderable-field-utils';
+import { iGet } from '../../../../utils/immutable';
+
+const ProvisionSourceDiskHelpMsg: React.FC<ProvisionSourceDiskHelpMsgProps> = ({
+  provisionSourceValue,
+  goToStorageStep,
+}) => {
+  const storageBtn = (
+    <Button
+      isDisabled={!goToStorageStep}
+      isInline
+      onClick={goToStorageStep}
+      variant={ButtonVariant.link}
+    >
+      <strong>Storage</strong>
+    </Button>
+  );
+  const getStorageMsg = () => {
+    switch (provisionSourceValue) {
```

Comment:
might be easier to call fromString once than toString 3times ..

Reply:
done